addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1742,6 +1742,10 @@ options.registry.CarouselItem = options.Class.extend({
                 this.$controls.toggleClass('d-none', newLength === 1);
                 this.$carousel.trigger('content_changed');
                 this.removing = false;
+                const indicatorsEls = this.$indicators[0].querySelectorAll('li');
+                for (let i = 0; i < indicatorsEls.length; i++) {
+                    indicatorsEls[i].setAttribute('data-bs-slide-to', i);
+                }
```

Comment:
Commit message well structured but I don't understand how the commit you mentioned is the source of the error. And your diff seems to make sense but how did it work before ? :thinking:

Reply:
> Before the commit I mentionned, [this line](https://github.com/odoo/odoo/commit/f7055d3dbabfbe471f490bd65c2032f5251f3f37#diff-70f7fe38208aa7fe678f18e329d3c11b70065dee723921352b6005774e8bab53L1690) removed the last <li data-slide-to="X"> and so there was indeed for each slide an indicator with the slide index.

Indeed :+1: 
- Why don't mention that in the commit message directly?
- Why don't you restore that instead of your new diff which requires to write on all indicators?

> But if you consider that the bug ARD fixed in the commit I mentioned should also be fixed in 14.0, this commit should also be on 14.0.. I think we should indeed do that

I consider that bugs should be fixed where they occur, as always. I don't think ARD's commit is fixing a bug that was there before the iframe edition? What is the bug in 14.0 you mention?

Reply:
> Why don't mention that in the commit message directly?

I will :+1: 
> Why don't you restore that instead of your new diff which requires to write on all indicators?

ARD told me that the purpose of the changes he mades in the removeSlide function of the commit I mentioned is to avoid another indicator issue. AFAIU the issue was:
- Drop a carousel
- Remove a slide

=> there is no active indicator.  Here is a video of the issue https://youtu.be/GHv6puVZ9bw 

Let's make his diff + my diff in 14.0 if it's ok for you

Reply:
Ok indeed :+1: @detrouxdev should have made a fix in stable rather than hide it in its big commit then :+1: So yes retarget this PR to 14.0 and make the proper complete fix. Also:

- Add comments in the code about that indicator removal of the active one + re-attribution of indexes
- Make the reattribution of indexes just after the removal... especially important to make it before the trigger of `content_changed` because listening code should be able to rely on the fact the carousel is in a OK state once that event is received.